File: 206-reverse-linked-list/reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode temp=null;
     while(head!=null){
          ListNode newnode = new ListNode(head.val);
          newnode.next=temp;
          temp=newnode;
          head=head.next;
     }  
     return temp;
    }
}

File: 206-reverse-linked-list/reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode *result=NULL ;
    if(head==NULL || head->next==NULL)
    {
        result=head ;
        return result ;
    }
    while(head!=NULL)
    {
        struct ListNode *copy ;
        copy=head ;
        head=head->next ;
        copy->next=result ;
        result=copy ;
    }
    return result ;
}